fpm/src/fpm.f90

```diff
@@ -4,8 +4,8 @@ module fpm
 use fpm_backend, only: build_package
 use fpm_command_line, only: fpm_build_settings, fpm_new_settings, &
                       fpm_run_settings, fpm_install_settings, fpm_test_settings
-use fpm_environment, only: run, get_os_type, OS_LINUX, OS_MACOS, OS_WINDOWS
-use fpm_filesystem, only: is_dir, join_path, number_of_rows, list_files, exists, basename, mkdir
+use fpm_environment, only: run 
+use fpm_filesystem, only: is_dir, join_path, number_of_rows, list_files, exists, basename 
```

Comment:
```suggestion
use fpm_environment, only: run
use fpm_filesystem, only: is_dir, join_path, number_of_rows, list_files, exists, basename
```

Reply:
I initially put it in a subdirectory called subcommands as I was thinking build/run/test have interdepenencies; but commands like search and other "missing" commands will probably be much better of with it refactored into seperate commands. Since it is likely to be used in only the places where it is now (and does not have to be typed over and over in code) I actually liked a longer descriptive name though.  I think I will put the subdirectory back. As you mentioned that is likely to happen later. But I sort of liked cmd_name, and everything seemed to have an fpm_ prefix, so how about
/subcommand/fpm_cmd_new.f90  with module fpm_cmd_new and leave the procedure called cmd_new and meet in the middle?

Reply:
That was not quite as simple as I thought.  h-fpm has some rules about subdirectories and names I am not sure will exist in f-fpm so I did not make the subdirectory after all.

Reply:
There are trailing spaces in several files I was not changing. Do you want me to remove them as well? It would add several files to the PR, however.
[]$ grep -rl ' $' *|more
fpm/error.f90
fpm/manifest/build_config.f90
fpm/manifest/package.f90
fpm_backend.f90
fpm_command_line.f90
fpm_filesystem.f90
fpm_model.f90
fpm_sources.f90
fpm_strings.f90
```

```text

Reply:
No need to consider the other files with trailing spaces in this PR.